ros/src/inertial_sense_ros.cpp

```diff
@@ -535,21 +535,36 @@ bool InertialSenseROS::connect(float timeout)
 {
     uint32_t end_time = ros::Time::now().toSec() + timeout;
     auto ports_iterator = ports_.begin();
+    auto serialNum = 0;
 
     do {
         std::string cur_port = *ports_iterator;
         /// Connect to the IMX
         ROS_INFO("InertialSenseROS: Connecting to serial port \"%s\", at %d baud", cur_port.c_str(), baudrate_);
         sdk_connected_ = IS_.Open(cur_port.c_str(), baudrate_);
-        if (!sdk_connected_) {
+        if (sdk_connected_)
+        {
+            serialNum = IS_.DeviceInfo().serialNumber;
+        }
+        
+        if (!sdk_connected_) 
```

Comment:
This logic feels fishy.  This should probably be an `} else {` of the previous if, since its the only other possible result of `sdk_connected_`).
Then the following `if (serialNum == 0)` should be inside the 'if connected" stanza.  If you are NOT connected, because `IS_.Open()` failed, the only thing to do is print the error, and move on.

Reply:
I added this piece which, yes does seem odd, but it helps identify a false connection which is indicated by a serial number of 0.

Reply:
I think this probably poor design then (somewhere, if not here).  While not "normal", its possible to have a 0 serial number.  If IS_.Open() is returning true/success, but serialNum == 0, then that implies some other problem (open port but unresponsive device, or failed to query devInfo, etc).  I think, within the `develop` branch, there are probably better ways to check for a valid device.  Check out the ISDevice class; there is a hdwRunState, that maybe more reliable.

In any case, this is probably better than nothing at all.